### PR TITLE
5475 - UI fixes based on feedback (mostly validation)

### PR DESCRIPTION
**Pull request by jordiwes**
*Issue #:* /bcgov/entity#5475

*Description of changes:*

- validation for document delivery section
- validation for certify section
- ui updates and alignments

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
   "dependencies": {
     "@babel/compat-data": "^7.11.0",
     "@bcrs-shared-components/action-chip": "1.0.2",
-    "@bcrs-shared-components/certify": "1.0.7",
+    "@bcrs-shared-components/certify": "1.0.11",
```
Comment (severinbeauvais) [inline package.json:17]: Please bump up the app version too.

File: src/components/Summary/DocumentsDelivery.vue [hunk 2]
```diff
@@ -39,21 +27,32 @@
               label="Optional Email"
               hint="Example: name@email.com"
               persistent-hint
+              validate-on-blur
```
Comment (severinbeauvais) [inline src/components/Summary/DocumentsDelivery.vue:30]: So does this work as expected -- the validation rules are not evaluated until the user leaves this text field?

File: package.json (hunk 1 above)
Comment (jordiwes) [reply package.json:17]: Always forget that. Done.

File: src/components/Summary/DocumentsDelivery.vue (hunk 2 above)
Comment (jordiwes) [reply src/components/Summary/DocumentsDelivery.vue:30]: Yep!

File: src/views/Alteration.vue [hunk 3]
```diff
@@ -48,12 +48,14 @@
 
       <documents-delivery
        class="mt-10"
+       :validate="getAppValidate"
        :pleaseValidate="true"
```
Comment (severinbeauvais) [inline src/views/Alteration.vue:52]: Is this prop now obsolete?

Also on line 58.

File: tests/unit/DocumentDelivery.spec.ts [hunk 4]
```diff
@@ -66,15 +67,14 @@ describe('Document Delivery component', () => {
 
     // Set Input field values
     vm.$el.querySelector(optionalEmailInput).textContent = '1212'
-    wrapper.find(optionalEmailInput).setValue('1212')
-    wrapper.find(optionalEmailInput).trigger('change')
-    await flushPromises()
+    wrapper.find('.text-input-field').trigger('focus')
+    wrapper.find('.text-input-field').find('input').setValue('1212')
+    wrapper.find('.text-input-field').trigger('blur')
```
Comment (severinbeauvais) [inline tests/unit/DocumentDelivery.spec.ts:72]: Nice!

File: src/views/Alteration.vue (hunk 3 above)
Comment (jordiwes) [reply src/views/Alteration.vue:52]: So, when the file and pay button is clicked, getAppValidate is set to true, triggering the component validation. Cam set it up for the court order stuff

Comment (severinbeauvais) [reply src/views/Alteration.vue:52]: What about `pleaseValidate` ?

Comment (jordiwes) [reply src/views/Alteration.vue:52]: That is hard coded to true and does nothing yet :) Should I remove?

Comment (severinbeauvais) [reply src/views/Alteration.vue:52]: Yes, please remove this. It was the initial validation implementation, now replaced by `getAppValidate`.

Comment (jordiwes) [reply src/views/Alteration.vue]: Done

**Review by severinbeauvais — APPROVED** [review]

**Review by cameron-freshworks — APPROVED** [review]

**Pull request merged**